src/pydsl/triton.py

```diff
@@ -0,0 +1,235 @@
+import ast
+import subprocess
+import typing
+from collections.abc import Iterable
+
+import triton
+from triton.backends.compiler import GPUTarget
+from triton.runtime.jit import JITFunction
+
+import mlir.ir as mlirir
+from mlir.ir import InsertionPoint
+from mlir.dialects import func
+
+from pydsl.type import Bool, Float, Index, Int, Number, Sign, UInt32
+from pydsl.memref import MemRef
+from pydsl.protocols import lower_single
+from pydsl.compiler import ToMLIRBase
```

Comment:
```suggestion
from pydsl.protocols import lower_single, ToMLIRBase
```
`ToMLIRBase` is defined in `protocols.py`.